package.json

```diff
@@ -4,19 +4,18 @@
   "main": "index.js",
   "scripts": {
     "start": "NODE_ENV=production node index.js",
-    "test:ci": "echo 'no tests for unleash-docker';"
+    "test:ci": "echo 'no tests for unleash-docker';",
+    "dev": "docker-compose up --build",
+    "dev:google": "docker-compose  -f docker-compose.yml -f docker-compose.google.yml up --build"
   },
   "engines": {
     "node": ">=12"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@exlinc/keycloak-passport": "^1.0.2",
```

Comment:
Why are you removing these? They were added to make this Docker image just work out of the box with a variety of OAuth2 providers.

Reply:
There is a not about why in the description

Reply:
Ah, I see the note now. The reason I added those dependencies is because we couldn't use any auth providers with the official Docker image at the time. Anyone who wanted to use Unleash with authentication had to build a custom image. It was a bad experience. What's the point of having a Docker image that no one [who uses auth] can use? I think we should either leave the dependencies in there to support multiple auth providers, or, if we're concerned about additional dependencies, we could create multiple Docker images. One barebones image, one for basic-auth, one for Google OIDC, one for Keycloak, etc. A different image and index.js for every auth provider out there...

Being a lazy person, I vote that we just leave all the dependencies in the official image so people can use Unleash without having to dig through documentation to figure out how to build their own custom images. Or maybe we could keep it simpler and just have two images: one with everything that just works, and one with nothing for ultimate security/simplicity.

Thoughts?

Reply:
Well, IMO adding dependencies that aren't used by the image, so that people can replace the index.js file is not a sustainable practice.

* What if I want to use a different/conflicting dependency, or different versions? That leads to multiple parties fighting for official versions. 
* Additionally, it leads to unnecessary potential vulnerabilities, by including dependencies that not everyone needs. Some of those dependencies are not what I would have expected as well. EG [the google oauth](https://github.com/passport-next/passport-google-oauth2#readme) one is from a fork with 5 stars, and is 95 commits behind the master.

In the case where you want custom oauth providers, I would actually recommend, just rolling your own service or docker image. 

I only added google in this PR, but it would be nice to support a bunch of providers by default, and keep the passport / oauth strategies up to date.

Reply:
So, what do we want to do then?
1. Have an opinionated image that only supports Google despite it being a sketchy dependency?
2. Have an essentially useless minimalist image that doesn't support any auth, and force everyone to create their own custom images and index.js files?
3.Have two images: one minimalist image, and one with everything to help people get proofs of concept up and running quickly and easily?
4. Have separate images for every auth provider?
I think #3 is a good compromise. If you rip out features and force customers to build custom images it's going to hinder adoption and ruin the customer experience.

Reply:
Just removing everything but Google is not cool. OIDC is a protocol, not a service - if we're going to have it in the product, we should support ALL OIDC providers.

Reply:
> 1. Have an opinionated image that only supports Google despite it being a sketchy dependency?

The old dependency is sketchy, i switched to a better supported one

> 2. Have an essentially useless minimalist image that doesn't support any auth, and force everyone to create their own custom images and index.js files?

That is more annoying than anything. It's like a bait an switch

> 3.Have two images: one minimalist image, and one with everything to help people get proofs of concept up and running quickly and easily?

That doesn't seem necessary, just have one that gracefully degrades when no environment variables are provided. Which is what this PR does.

> 3. Have separate images for every auth provider?
Thats not terrible TBH, but seems more annoying than anything.

> Just removing everything but Google is not cool. OIDC is a protocol, not a service - if we're going to have it in the product, we should support ALL OIDC providers.

IMO we should have a default setup for the most common ones a business would use:
Github, google, microsoft?, maybe facebook (would that be something you would want to use??)

We could also add http://www.passportjs.org/docs/openid/ for the generic case.